### PR TITLE
Close StringTable and BytesIO

### PR DESCRIPTION
**Pull request by twoertwein**
This should take care of `ResoruceWarning`s.

### CODE REVIEW

File: pyxlsb/workbook.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def _parse(self):
         temp.seek(0, os.SEEK_SET)
       self.stringtable = StringTable(fp=temp)
     except KeyError:
-      pass
+      temp.close()
```
Comment (willtrnr) [inline pyxlsb/workbook.py:56]: That's a good catch. However, calling the `close` only for `KeyError` is too restrictive, many other things can go wrong here.

Maybe add a bare-except to close and then reraise the exception.

**Review by willtrnr — CHANGES_REQUESTED** [review]
Good catches, this looks mostly fine. I'll release a patch version for this afterwards.

File: pyxlsb/reader.py [hunk 2]
```diff
@@ -74,6 +74,12 @@ def read_string(self):
       return None
     return buff.decode(self._enc, errors='replace')
 
+  def __enter__(self):
+    return self
+
+  def __exit__(self, type, value, traceback):
+    self._fp.close()
+
```
[the remark this answers is not in the document]
Comment (twoertwein) [reply pyxlsb/reader.py]: okay, done

File: pyxlsb/workbook.py (hunk 1 above)
Comment (twoertwein) [reply pyxlsb/workbook.py:58]: done

File: pyxlsb/workbook.py [hunk 3]
```diff
@@ -54,6 +54,9 @@ def _parse(self):
       self.stringtable = StringTable(fp=temp)
     except KeyError:
       pass
```
Comment (willtrnr) [inline pyxlsb/workbook.py:56]: Last thing, you also need to close on `KeyError` as well.

**Review by willtrnr — CHANGES_REQUESTED** [review]

**Comment by twoertwein** [comment]
> I'll release a patch version for this afterwards.

In case you are not waiting for any other PRs :)

**Comment by willtrnr** [comment]
I ended up being busy with other things, thanks for the reminder.

I published 1.0.8 right now with your patch.

**Pull request merged**